### PR TITLE
add domain filter option

### LINKED ISSUE #24: Add option to set domain/urls where the extension should work on

### PR DESCRIPTION
**Pull request by axellorreyne**
This fixes #24

### CODE REVIEW

**Comment by pheyvaer** [comment]
@axellorreyne Can you first fix the conflicts before I review? Thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Package instructions
 - Add instructions in README where people can find signed extension (see [issue 22](https://github.com/KNowledgeOnWebScale/solid-authentication-browser-extension/issues/22))
 - Mention the minimum version of CSS that is needed (see [issue 23](https://github.com/KNowledgeOnWebScale/solid-authentication-browser-extension/issues/23))
+- Domain filter (with optional regex syntax)
```
Comment (pheyvaer) [inline CHANGELOG.md:15]: Mention issue https://github.com/KNowledgeOnWebScale/solid-authentication-browser-extension/issues/24

File: src/popup/popup.js [hunk 2]
```diff
@@ -10,6 +12,11 @@ function main() {
     document.getElementById("email-input-form").addEventListener("keypress", submitLoginOnKeyEnter);
     document.getElementById("password-input-form").addEventListener("keypress", submitLoginOnKeyEnter);
     document.getElementById("idp-input-form").addEventListener("keypress", submitLoginOnKeyEnter);
+    document.getElementById("domain-input-form").addEventListener("keypress", submitLoginOnKeyEnter);
```
Comment (pheyvaer) [inline src/popup/popup.js:15]: Why is this only one field? What if a user wants to add multiple domains? Is it expected that they all add it in a single regex?

File: src/js/background.js [hunk 3]
```diff
@@ -4,6 +4,8 @@ import {createDpopHeader} from '@inrupt/solid-client-authn-core';
 var id;
 var secret;
 var tokenUrl;
+var domainFilter;
```
Comment (pheyvaer) [inline src/js/background.js:8]: Please have a look at [this](https://codeburst.io/javascript-var-let-or-const-which-one-should-you-use-2fd521b050fa) regarding the use of `var`.

**Comment by axellorreyne** [comment]
> @axellorreyne Can you first fix the conflicts before I review? Thanks!

Conflicts resolved in 4f06fb07bf0e7e9e0e1ccbc2e12869f8b5be1515

**Review by pheyvaer — CHANGES_REQUESTED** [review]
After logging in without domain filter and I open the pop up again I get this error:
image

**Comment by axellorreyne** [comment]
> After logging in without domain filter and I open the pop up again I get this error: image

This is caused by a bug on main, which was merged into this branch after resolving the aforementioned conflict. This issue has been fixed in #32 which was merged into main after the conflict resolution on this branch.

**Comment by pheyvaer** [comment]
Great! Can you merge `main` in this branch as well?

File: CHANGELOG.md (hunk 1 above)
Comment (axellorreyne) [reply CHANGELOG.md]: Added in c55cf1c404d5cb89e58bc683f24315792364309f

File: src/js/background.js (hunk 3 above)
Comment (axellorreyne) [reply src/js/background.js]: Changed in 25cb8895125bd8e147dbcf163b18068fd1029de4

**Comment by axellorreyne** [comment]
> Great! Can you merge `main` in this branch as well?

merged in ae860bc6724b8b80dd9fe1000a402dd18805dcd0

File: src/popup/popup.js (hunk 2 above)
Comment (axellorreyne) [reply src/popup/popup.js:18]: Having multiple input forms for different domain filters would make the popup too cluttered.
In the long term, I am planning to move all configurations to an [option page](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/user_interface/Options_pages) where we could also implement the use of multiple Client ID's and multiple domain filters per Client ID in a more structured manner.

**Review by pheyvaer — APPROVED** [review]

**Pull request merged**